### ISSUE TITLE
Move overscroll-behavior spec from WICG to csswg-drafts

### ISSUE THREAD

**Issue by majido** [w3c/csswg-drafts#2179]
As per earlier csswg [decision](https://github.com/w3c/csswg-drafts/issues/769#issuecomment-293111694) `overscroll-behavior` [specification](https://wicg.github.io/overscroll-behavior/) has been incubating in WICG for a while. The property has now shipped in [Chrome M63](https://www.chromestatus.com/features/5734614437986304) and it is planned to be shipped in [Firefox M59](https://groups.google.com/forum/#!topic/mozilla.dev.platform/6fVRnF4_g8s). The proposal also has [general approval](https://github.com/w3ctag/design-reviews/issues/193#issuecomment-346964139) of  TAG.

I propose to move the specification to csswg-drafts so that it can be tracked here and moves forward
in the standardization process.

**Comment by majido** [comment]
@bgirard @theres-waldo

**Comment by frivoal** [comment]
If it's already shipping (or almost) in two browsers, I'd actually say that we're quite late in taking it out of incubation.

The current editor, @bgirard, isn't a member of the CSSWG. Does he plan to join? Do we need to find/appoint a new editor?

**Comment by majido** [comment]
@bgirard can answer that but I thought Facebook is a member organization.
If it becomes necessary I can become a coeditor as well.

**Comment by frivoal** [comment]
I was no so much concerned about IP rights, which is what matters for whether or not Facebook is a members. Someone will have to care about that, but not me :)

No, I was just thinking of actually editing the document. It seems much simpler to have the editor going forward either be the same person that has been doing it until now, or at least someone who's been involved since the start. First it's just fair, there's not reason to take the work away from it's authors, and also we'll save a lot of time if the person handling it knows why things are the way they are.

So yes, if @bgirard can join the group, that'd be great, and if not, if you could, that'd great too.

But anyway, editorship is the chairs' call.

**Comment by svgeesus** [comment]
> @bgirard can answer that but I thought Facebook is a member organization.

Facebook is a member of W3C, but they have not joined the CSS Working Group. Each organization that is a W3C member has a designated contact point (AC member, in W3C jargon) so I sent a mail just now to the Facebook contact, inviting Facebook to join the CSS WG.

**Comment by bgirard** [comment]
That's correct. I'll follow-up as well with Facebook's W3C representative. Facebook had already began discussing joining so I hope finalizing that will be straight forward.

I'm happy to continue to edit the spec to the best of my ability, or to step aside if someone better suited wants to take over. As long as we have a good quality spec in sync with the current implementations and vise versa.

**Comment by majido** [comment]
Great. It seems that we should hopefully be able to have @bgirard continue as the editor if chair does not object.

Setting that question aside, I see now that this is scheduled to be on the agenda. So I believe we just wait for WG to approve.

I just wanted to add a few minor notes/questions on the migration process: 

- The spec in already in bikeshed so it should not need a lot of formatting change. But it may be valuable to have a csswg expert to do a review to ensure we are doing the right thing e.g., linking to the appropriate concepts etc. Probably best to happen after migration.
- There are two small open issues. It will be nice if we close them before migration but otherwise we need to migrate those as well.
- I know of at least one specification, [scroll-anchoring](https://drafts.csswg.org/css-scroll-anchoring/), that has gone from wicg->csswg-draft. So WICG folks may already have a well established process. Here is more [details](https://github.com/w3c/csswg-drafts/issues/676#issuecomment-296109835).
- Is there a way to keep commit history remain intact? It will be nice but I am not sure how given that this is not just a matter or moving the repo across organizations. This is not critical though since the spec is not very old.

**Comment by birtles** [comment]
> * Is there a way to keep commit history remain intact? It will be nice but I am not sure how given that this is not just a matter or moving the repo across organizations. This is not critical though since the spec is not very old.

Yes, there is. I moved web-animations to csswg-drafts recently and _after_ I did that @SimonSapin  told me how to do it:

> "What I’ve done before in other repositories is make a commit that moves everything into a sub-directory, and then merge the two previously-unrelated branches. That way history is preserved without rewriting commit hashes."

@gsnedders also has experience with this.

**Comment by tantek** [comment]
+1 let's move this spec to CSSWG please. I'd prefer to see editorship continuity if at all possible. Thanks!

**Comment by fantasai** [comment]
Agenda+ because it seems that nobody followed up on this, and somebody needs to.

**Comment by astearns** [comment]
@bgirard any update on having Facebook join the working group? We tried to contact your AC rep but did not get a response.

**Comment by bgirard** [comment]
I chatted about this with our AC rep and a few other folks and I'm not sure if I can commit enough time to join the CSS working group and I haven't been able to find someone else to take it on right now 😕.

That being said, I'm hoping that as we on board more people to our team we will be able to find someone to participate in the group in the future.

**Comment by svgeesus** [comment]
@bgirard the amount of time you commit is largely up to you. It is fine to have only one spec you are interested in, for example. 

Would it be possible to have you join now, and then transition that role to someone else as your time grows in the future?

**Comment by majido** [comment]
@bgirard  I had a chat with @astearns and a few others at CSSWG F2F yesterday. I am planning to join the WG as a member mainly to continue animation worklet specification as it moves to the working group. I had concerns around how much time I need to spend particularly for the weekly teleconferences.

He confirmed that the expectation for editors of specs with limited scope is to simply attend such meetings only when there is an agenda item related to the specification. Agenda is posted in advance of the meeting which helps decide if attendance is needed. Most of the issues on specs don't need to be discussed on teleconference and will be resolved via github.

This helped convince me that there is limited overhead in my case. Just wanted to share this in case it helps with your decision one way or another.

**Comment by majido** [comment]
@bgirard has there been any change in your plans regarding joining the CSSWG?

Unless we have new updates, I think we should follow with our plan as discussed in F2F and noted [above](https://github.com/w3c/csswg-drafts/issues/2179#issuecomment-380025995) and move the repo to CSSWG.

**Comment by majido** [comment]
I will handle the move logistics. Here is my plan so far, please make suggestions and corrections as you see fit.

## Merging repos without losing history

I did a test run based on @birtles  [suggested path](https://github.com/w3c/csswg-drafts/issues/2179#issuecomment-357089856) and the history remains intact.

Basically, it is a merging two unrelated git histories but git is happy to do so none the less and things seems to be working as expected.

Here is the series of command I used:

``` shell
$ cd csswg-drafts
$ git remote add wicg git@github.com:WICG/scroll-boundary-behavior.git
$ git fetch wicg
$ git checkout -b wicg wicg/master
$ mkdir css-overscroll-behavior
$ mv * css-overscroll-behavior/
$ git add .
$ git commit -m "Move to sub-directory in preparation to move the repository"
$ git checkout master 
$ git merge wicg
$ git merge --allow-unrelated-histories wicg
```

I pushed the result to my GH fork and all the history is intact. See [here](https://github.com/majido/csswg-drafts/blame/merge-ob-repo/css-overscroll-behavior/index.bs) for example.

I suspect then it is just a matter of sending a PR for the above repr and making sure the PR is merged without squashing and rebasing. 

## Migrating open issues
There are three [open issues](https://github.com/WICG/overscroll-behavior/issues). I will manually re-create these in csswg-drafts with proper labels etc.

## Post move

1. Update index.bs fix
    - shortname to `css-overscroll-behavior`
    - URL to `https://drafts.csswg.org/css-overscroll-behavior/`
    - Status to `ED` 
    - Repository to `https://github.com/w3c/csswg-drafts/tree/master/css-overscroll-behavior css-overscroll-behavior`
    - Add myself as an editor
2. Remove LICENSE.md, CONTRIBUTING.md, security-privacy-questionnaire.md

The spec is already in Bikeshed and builds without any warnings so hopefully that means it will not create any issues for csswg-draft tools.
So my plan is to merge the repo as is (with the exception of moving to subdirectory) and make necessary changes to deal with any issues after the merge. Let me know if there is anything I am missing or should try to fix before merging the repo.

**Comment by ewilligers** [comment]
[https://drafts.csswg.org/css-overscroll-behavior/](https://drafts.csswg.org/css-overscroll-behavior/) currently gives a list of files.
[https://drafts.csswg.org/css-overscroll-behavior/index.html](https://drafts.csswg.org/css-overscroll-behavior/index.html) currently gives the spec.

For most specs, `/` gives the spec and `/index.html` gives `404 Not Found`.

**Comment by frivoal** [comment]
@ewilligers, fixed https://github.com/w3c/csswg-drafts/commit/e6103a5efe9b1cc529ba97270d00119535a1c970

**Comment by frivoal** [comment]
I think we just need @majido to add himself as an editor, and then we can close this.

**Comment by xfq** [comment]
And we probably want to publish a FPWD too.

**Comment by majido** [comment]
@frivoal Sent a PR to add myself as an editor.
@xfq I agree. I just added an agenda item for TPAC F2F to discuss this spec and move it to FPWD.

**Comment by frivoal** [comment]
@majido Merged, but you don't need pull requests for changes like this. You're an editor, you can commit directly. Pull requests are useful for proposing changes to the WG prior to getting a resolution, or for asking someone to check after implementing a subtle resolution, but generally fell free to just commit directly to the spec. If you're a bit unsure about the balance of power between the Editor and the WG in the CSSWG, [here is a good reference](http://fantasai.inkedblade.net/weblog/2011/inside-csswg/decisions).
